api/paidAction/index.js

```diff
@@ -237,39 +263,60 @@ async function createLndInvoice (actionType, args, context) {
   })
 }
 
-async function createDbInvoice (actionType, args, context) {
-  const { me, models, tx, lndInvoice, cost, optimistic, actionId } = context
+async function createDbInvoice (actionType, args, context,
+  { invoice: unwrappedInvoice, wrappedInvoice, wallet, maxFee }) {
+  const { me, models, tx, cost, optimistic, actionId } = context
   const db = tx ?? models
-  const [expirePivot, actionState] = optimistic
-    ? [OPTIMISTIC_INVOICE_EXPIRE, 'PENDING']
-    : [PESSIMISTIC_INVOICE_EXPIRE, 'PENDING_HELD']
 
   if (cost < 1000n) {
     // sanity check
     throw new Error('The cost of the action must be at least 1 sat')
   }
 
-  const expiresAt = datePivot(new Date(), expirePivot)
-  const invoice = await db.invoice.create({
-    data: {
-      hash: lndInvoice.id,
-      msatsRequested: cost,
-      preimage: optimistic ? undefined : lndInvoice.secret,
-      bolt11: lndInvoice.request,
-      userId: me?.id ?? USER_ID.anon,
-      actionType,
-      actionState,
-      actionArgs: args,
-      expiresAt,
-      actionId
-    }
-  })
+  const servedInvoice = wrappedInvoice ?? unwrappedInvoice
+  const request = parsePaymentRequest({ request: servedInvoice.request })
```

Comment:
Mhh, I tested zapping someone with an attached wallet and without since I noticed that `unwrappedInvoice` is just a bolt11 in the wrapped case and wondered how `parsePaymentRequest` is going to work with `servedInvoice.request` being undefined in the unwrapped case.

Turns out it's not undefined since `unwrappedInvoice` is the return value of a LND call and not just the bolt11 in that case. So this code does not contain a bug but it's unexpected that the type of a variable might be different depending on context.

A comment might be useful here since I was also confused about `bolt11: unwrappedInvoice` in the `db.invoiceForward.create` call.

```suggestion
  // if we wrapped the invoice, the unwrapped invoice is a bolt11 string else it's a full lightning invoice object
  const servedInvoice = wrappedInvoice ?? unwrappedInvoice
  const request = parsePaymentRequest({ request: servedInvoice.request })
```

Reply:
I'll take a closer look at this. The fact that I'm not sure what these types are is a sign this should be changed.